### ISSUE TITLE
Get rid of log spam `access denied to Proxy`

### ISSUE THREAD

**Issue by klizhentas** [gravitational/teleport#2920]
## Description

We get log spam on teleport side that confuses users:

```
Access to create user in namespace default denied to roles Proxy,default-implicit-role: no allow rule matched.
```

**Comment by fspmarshall** [comment]
Are we trying to reduce the spam of this message in general (e.g. move it from `INFO` to `DEBUG` level), or is there a specific condition that is currently emitting this message when it shouldn't?

**Comment by klizhentas** [comment]
@fspmarshall lets start with building a good understanding of why this message is emitted in this case, once you post it here we can come up with a plan

**Comment by fspmarshall** [comment]
@klizhentas alright.  Do you have any context for what configurations/actions have lead to this log being spammed historically?  Would be helpful to be able to replicate the issue.

**Comment by klizhentas** [comment]
AFAIK if you use docker-compose cluster, you should see the warnings right away

**Comment by fspmarshall** [comment]
Fairly certain this was addressed in #3232.  If this issue crops up again we can re-open.

**Comment by webvictim** [comment]
Reopening this as the PR with the fix hasn't been merged yet.
